features/steps/steps.py

```diff
@@ -413,3 +485,64 @@ def step_impl(context, entity, other_entities):
                 errors.append(instance_structure_error(inst, [i for i in nested_entities if i.is_a() in differences], 'nested by'))
     
     handle_errors(context, errors)
+
+def get_duplicate_values(values):
+    seen = set()
+    duplicates = [x for x in values if x in seen or seen.add(x)]
+    return duplicates
+
+def ifcopenshell_instance_type_to_string(v):
+    """
+        Converts ifcopenshell instance type to strings, if applicable
+    """
+    return do_try(lambda: v[0].is_a(), v)
+
+def empty_tuple_to_string(v):
+    """
+        Converts empty tuples type to strings, if applicable
+        To be used for meaningful error messages
+    """
+    return 'None' if isinstance(v, tuple) and not v else v
+
+def map_many(v, fn, *args):
+    """
+        Maps multiple functions to a list
+    """
+    return map_many(map(fn, v), *args) if args else map(fn, v)
+
+@then("The values must be {identical_or_unique}")
+def step_impl(context, identical_or_unique):
+    errors = []
+
+    if getattr(context, 'applicable', True):
+
+        stack_tree= list(filter(None, list(map(lambda layer: layer.get('instances'), context._stack))))
+
+        for i, values in enumerate([context.instances] if getattr(context, 'within_model', False) else context.instances):
+            if not values:
+                continue
+                        
+            values_str = list(map_many(values, empty_tuple_to_string, ifcopenshell_instance_type_to_string))
+            attribute = getattr(context, 'attribute', 'None')
+
+            duplicates = get_duplicate_values(values_str)
+
+            if (identical_or_unique == 'identical' and len(values_str) > 1 and not duplicates):
```

Comment:
I don't think `len(values_str) > 1 and not duplicates` is correct. `['a', 'a', 'b']` implies `len(values_str) == 3`, but `duplicates == ['a']` so not empty. I think it can only be `len(set(values_str)) == 1`. Knowing only the duplicates and the number of original values is not sufficient.

Reply:
You're right. I've added an extra testfile with values ['IfcMapConversion', 'IfcRigidOperation', 'IfcMapConversion'] and this leads to a false positive. 
If I'm correct, len(set(values)) will fix this, i.e.
```
>>> aa = ['a', 'a']
>>> len(set(aa)) > 1
False # function will continue
>>> ab = ['a', 'b']
>>> len(set(ab)) > 1
True # generate error
>>> aab = ['a', 'a', 'b']
>>> len(set(aab)) > 1
True # generate error
>>> a = ['a']
>>> len(set(a)) > 1
False # continue function
```
Added in https://github.com/buildingSMART/ifc-gherkin-rules/commit/ab11bfef554cdb739d2beb505f2c0e99b80362bb